pkg/sources/http/http.go

```diff
@@ -256,7 +265,7 @@ func (h *httpSource) Close() error {
 
 func (h *httpSource) Stop() {
 	h.logger.Info("Stopping http reader...")
-	h.ready = false
+	defer func() { h.ready = false }()
```

Comment:
Why let it execute in the end?

Reply:
Should we mark httpSource as not ready **after** instead of **before** we stop the forwarder?

Reply:
No, marking it not ready beforehand is an intended behavior.

Marking it not ready will get the health check unsuccessful, and bring it out of the traffic (if there's LB existing). Imagine what will happen that you are already in the process of shutting down backend services, but the traffic is still coming in.